testsprite_tests/TC004_Shipment_Management___Create_New_Shipment_with_Valid_Data.py

```
@@ -0,0 +1,130 @@
+        frame = context.pages[-1]
+        # Click on 'New Shipment' button to open shipment creation form
+        elem = frame.locator('xpath=html/body/div[2]/main/main/div/div/div[2]/a').nth(0)
+        await page.wait_for_timeout(3000); await elem.click(timeout=5000)
+        
+
+        # --> Assertions to verify final state
+        frame = context.pages[-1]
+        try:
+            await expect(frame.locator('text=Shipment Creation Failed').first).to_be_visible(timeout=1000)
+        except AssertionError:
+            raise AssertionError("Test case failed: Shipment creation did not succeed as expected. The shipment was not saved or displayed correctly in the shipment list, or server actions did not trigger cache revalidation and UI state updates as required by the test plan.")
```

Ruff (0.14.10)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)